### PR TITLE
Update picocli to 4.6.1

### PR DESCRIPTION
**Pull request by wendigo**
Picocli 4.6.0 adds support for Optional<T> which makes one of the type converters unnecessary. The other one is removed as well.

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentUp.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public Integer call()
         @Option(names = "--environment", paramLabel = "<environment>", description = "Name of the environment to start", required = true)
         public String environment;
 
-        @Option(names = "--logs-dir", paramLabel = "<dir>", description = "Location of the exported logs directory", converter = OptionalPathConverter.class, defaultValue = "")
+        @Option(names = "--logs-dir", paramLabel = "<dir>", description = "Location of the exported logs directory", defaultValue = "")
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentUp.java:91]: is defaultValue `""` applicable to Optional as well?
or will it result in using `Path.get("")` as the logs dir?

**Comment by wendigo** [comment]
AC @findepi

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@wendigo please check the build

**Comment by wendigo** [comment]
@electrum are you gonna take a look or it's ready to merge?

**Pull request merged**